driverinference/README.md

```diff
@@ -0,0 +1,301 @@
+
+FLAGS
+    --exp_name=EXP_NAME
+        Default: 'inference-colocated-scaling'
+        name of output dir, defaults to "inference-scaling"
+    --node_feature=NODE_FEATURE
+        Default: {'constraint': 'P100'}
+    --launcher=LAUNCHER
+        Default: 'auto'
+        workload manager i.e. "slurm", "pbs"
+    --nodes=NODES
+        Default: [4,8,16,32,64,128]
+        compute nodes to use for synthetic scaling app with a co-located orchestrator database
+    --clients_per_node=CLIENTS_PER_NODE
+        Default: [18]
+        client tasks per compute node for the synthetic scaling app
+    --db_cpus=DB_CPUS
+        Default: [8]
+        number of cpus per compute host for the database
+    --db_tpq=DB_TPQ
+        Default: [8]
+        number of device threads to use for the database
+    --db_port=DB_PORT
+        Default: 6780
+        port to use for the database
+    --batch_size=BATCH_SIZE
+        Default: [96]
+        batch size to set Resnet50 model with
+    --device=DEVICE
+        Default: 'GPU'
+        device used to run the models in the database
+    --num_devices=NUM_DEVICES
+        Default: 1
+        number of devices per compute node to use to run ResNet
+    --net_type=NET_TYPE
+        Default: 'uds'
+        type of connection to use ("tcp" or "uds")
+    --net_ifname=NET_IFNAME
+        Default: 'lo'
+        network interface to use i.e. "ib0" for infiniband or "ipogif0" aries networks
+    --iterations=ITERATIONS
+        Default: 100
+        number of put/get loops run by the applications
+    --languages=LANGUAGES
+        Default: ['cpp','fortran']
+        list of languages to use for the tester "cpp" and/or "fortran"
+    --plot=PLOT
+        Default: 'database_cpus'
+        flag to plot against in process results
+```
+
+So for example, the following command could be run to execute a battery of
+tests in the same allocation
+
+```bash
+# alloc must contain at least 16 GPU nodes
+python driver.py inference_colocated --clients_per_node=[18] \
+                                     --nodes=[16] --db_tpq=[1,2,4] \
+                                     --db_cpus=[1,2,4,8] --net_ifname="ipogif0" \
+                                     --device="GPU"
+```
+
+This command can be executed in a terminal with an interactive allocation
+or used in a batch script such as the following for Slurm based systems
+
+```bash
+#!/bin/bash
+
+#SBATCH -N 16
+#SBATCH --exclusive
+#SBATCH -C P100
+#SBATCH -t 10:00:00
+
+python driver.py inference_colocated --clients_per_node=[18] \
+                                     --nodes=[16] --db_tpq=[1,2,4] \
+                                     --db_cpus=[1,2,4,8] --net_ifname="ipogif0" \
+                                     --device="GPU"
+```
+
+Examples of batch scripts to use are provided in the ``batch_scripts`` directory
+
+
+## Standard Inference
+
+Co-located deployment is the preferred method for running tightly coupled
```

Comment:
I'll mention co-located needs to be changed to colocated here one last time, but please do this globally